test/user.test.js

```diff
@@ -2428,6 +2454,47 @@ describe('User', function() {
             .expect(200));
       });
 
+      it('should reject password reset role/user relations', function(done) {
+        var accessToken2, AdminUserReset;
+        async.series([
+          function(next) {
+            BasicUser.create({ name: 'BU', email: 'u@example.com', password: 'foobar'}, function(err, user) {
+              if (err) return done(err);
+              next();
+            });
+          },
+          function(next) {
+            AdminUser.create({name: 'AU', email: 'a@example.com', password: 'foobar'}, function(err, user) {
+              AdminUserReset = user;
+              if (err) return done(err);
+              next();
+            });
+          },
+          function(next) {
+            BasicUser.login({email:'u@example.com', password: 'foobar'}, function(err, accessToken1) {
+              accessToken2 = accessToken1;
+              if (err) return next(err);
+              assert(accessToken1.id);
+              next();
+            });
+          },
+          function(next) {
+            request(app)
+              .post('/test-basic/reset-password')
+              .set('Authorization', accessToken2.id)
```

Comment:
IIRC, the `reset-password` endpoint does not require authentication - how is the user supposed to provide a valid access token when they cannot remember their password needed to create one?

Reply:
Ah sorry, I got confused because there are two similar endpoints: `/reset` is called to issue a password-reset token, `/reset-password` is called to perform password reset.

Note that using a token created by `login` to reset a password is a design flaw that's supported for backwards compatibility only. Your test should be calling `BasicUser.resetPassword` instead of `BasicUser.login` to create the token.

I also think that in order to reproduce the problem you are trying to fix, there is no need to use REST API and you can invoke `BasicUser.setPassword`/`AdminUser.setPassword` instead.